examples/Example19-Interrupt/Example19-Interrupt.ino

```diff
@@ -0,0 +1,142 @@
+  and maintains a flag for new data of 2 different sensor types.
+
+  Hardware Connections:
+  Attach the Qwiic Shield to your Arduino/Photon/ESP32 or other
+  Plug the sensor onto the shield
+  Serial.print it out at 9600 baud to serial monitor.
+*/
+
+#include <Wire.h>
+
+#include "SparkFun_BNO080_Arduino_Library.h" // Click here to get the library: http://librarymanager/All#SparkFun_BNO080
+BNO080 myIMU;
+
+// indicators of new data availability
+volatile byte newQuat = 0;
+volatile byte newLinAcc = 0;
+// internal copies of the IMU data
+float ax, ay, az, qx, qy, qz, qw; // (qx, qy, qz, qw = i,j,k, real)
+byte linAccuracy = 0;
+float quatRadianAccuracy = 0;
+byte quatAccuracy = 0;
+// pin used for interrupts
+byte imuINTPin = 3;
+
+void setup()
+{
+  Serial.begin(9600);
+  Serial.println();
+  Serial.println("BNO080 Read example with Interrupt handler, getReading and multiple access");
+
+  Wire.begin();
+  // imuINTPin should be input and pulled-up to be used as active-low interrupt
+  pinMode(imuINTPin, INPUT_PULLUP);
+
+  //Are you using a ESP? Check this issue for more information: https://github.com/sparkfun/SparkFun_BNO080_Arduino_Library/issues/16
+//  //=================================
+//  delay(100); //  Wait for BNO to boot
+//  // Start i2c and BNO080
+//  Wire.flush();   // Reset I2C
+//  IMU.begin(BNO080_DEFAULT_ADDRESS, Wire);
+//  Wire.begin(4, 5);
+//  Wire.setClockStretchLimit(4000);
+//  //=================================
+
+  if (myIMU.begin(BNO080_DEFAULT_ADDRESS, Wire, imuINTPin) == false)
+  {
+    Serial.println("BNO080 not detected at default I2C address. Check your jumpers and the hookup guide. Freezing...");
+    while (1);
+  }
+
+  Wire.setClock(400000); //Increase I2C data rate to 400kHz
+
+  // prepare interrupt on falling edge (= signal of new data available)
+  attachInterrupt(digitalPinToInterrupt(imuINTPin), interrupt_handler, FALLING);
+  // enable interrupts right away to not miss first reports
+  interrupts();
+
+  myIMU.enableLinearAccelerometer(50);  // m/s^2 no gravity, data update every 50 ms
+  myIMU.enableRotationVector(100); //Send data update every 100 ms
+
+  Serial.println(F("LinearAccelerometer enabled, Output in form x, y, z, accuracy, in m/s^2"));
+  Serial.println(F("Rotation vector, Output in form i, j, k, real, accuracy"));
+
+}
+
+// This function is called whenever an interrupt is detected by the arduino
+void interrupt_handler()
+{
+  // code snippet from ya-mouse
+   switch (myIMU.getReadings())
+   {
+      case SENSOR_REPORTID_LINEAR_ACCELERATION: {
+        newLinAcc = 1;
+      }
+      break;
+
+      case SENSOR_REPORTID_ROTATION_VECTOR:
+      case SENSOR_REPORTID_GAME_ROTATION_VECTOR: {
+         newQuat = 1;
+      }
+      break;
+      default:
+         // Unhandled Input Report
+         break;
+   }
+}
+
+void loop()
+{
+  Serial.println("Doing other things");
+  delay(10); //You can do many other things. Interrupt handler will retrieve latest data
```

Comment:
Please pay attention, that BNO080 would be reset by it's watchdog if the interrupt wouldn't be de-asserted at the requested rate.

SparkFun has older datasheet for BNO080 (1.3).
The recent revision of the datasheet (1.8) says:
https://www.ceva-dsp.com/wp-content/uploads/2019/10/BNO080_085-Datasheet.pdf

```
On the BNO080, if the host fails to respond to the assertion of H_INTN, the BNO080 will eventually reset due to a watchdog timeout.

On the BNO085, if the host fails to respond to the assertion of H_INTN within approximately 10 ms, the BNO085
will timeout, deassert H_INTN and retry the operation.
```

Reply:
Should I add this information in the example ? because I don't know what the example is not doing correctly. It is just sleeping and doing some prints, and gets interrupted when data is there.  Do you think the interrupt handler is too slow on the arduinos to handle data transfer per interrupt ? 
I must be missing knowledge there.

I know I did not handle potential interrupts arriving while accessing the data in the main loop, potentially producing, inconsistent data.  Reading one data at a time will not help, I only see a way by doing a flip/flop double buffer, filling one, reading the other one but that became too much for just a small example.

Reply:
It might be worth to mention that in interrupt handler. The loop() is fine.

Reply:
Just to be sure, if one is polling, which all other examples do (if I am not mistaken), this problem will also occur when the main loop does not "de-assert" (or to use simpler word, "react" to ) the H_INTN fast enough. I even think it is worse when polling in the main loop because the main loop might really have some delay doing a "dataAvailable" call when the interrupt should be faster to react isn't it ?

Reply:
IMO, do not use loop() to de-assert H_INTN pin. Do this only within interrupt handler otherwise use I2C.
For I2C interface the BNO080 doesn't care if you're reading fast enough. That's only noted for SPI interface.

Reply:
Hi, @ya-mouse , I am sorry but I still don't get the point. The example is for I2C and now you say that the mention about de-asserting H_INTN fast enough is for SPI interface only. So why should I write this special mention in my example ? 
I am really confused on what you expect me to fix here.

Reply:
sorry for confusion, I just saw the interrupt handler and realized that it's about SPI.